bitnami/mongodb-sharded/templates/shard/shard-data-statefulset.yaml

```
@@ -97,6 +97,12 @@ spec:
             - containerPort: {{ $.Values.service.port }}
               name: mongodb
           env:
+            - name: MONGODB_ENABLE_NUMACTL
+              {{- if $.Values.common.mongodbEnableNumactl }}
+              value: "yes"
+              {{- else }}
+              value: "no"
+              {{- end }}
```

Please use [ternary](https://helm.sh/docs/chart_template_guide/function_list/#ternary) instead:

```suggestion
            - name: MONGODB_ENABLE_NUMACTL
              value: {{ ternary "yes" "no" $.Values.common.mongodbEnableNumactl }}
```

---

Also, I'm not familiar with the Numa features, but shouldn't this also be added to all the Statefulsets? (mongos, config, arbiter and data)

I see config and data statefulsets, but I'm missing `mongos` and `arbiter`.

From [MongoDB documentation](https://docs.mongodb.com/manual/administration/production-notes/#configuring-numa-on-linux), sounds like every mongod and mongos instances require this setting, right?
> Ensure that mongod and mongos are started by numactl.

---

You're right.
Added to arbiter and mongos and changed to ternary in config statefulset too.